compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -907,4 +907,28 @@ object JavaParsers {
       unit
     }
   }
+
+
+  /**
+   * OutlineJavaParser parses top-level declarations in `source` to find declared classes, ignoring their bodies (which
+   * must only have balanced braces). This is used to map class names to defining sources.
+   * This is necessary even for Java, because the filename defining a non-public classes cannot be determined from the
+   * classname alone.
+   */
+  class OutlineJavaParser(source: SourceFile)(implicit ctx: Context) extends JavaParser(source) {
```

Comment:
Nitpick: the documentation style we (try to) use across the code base is:
```scala
/** First line
 *
 *  More lines...
 *  @param ...
 */
def foo = 
```
We start on the first line and all lines are alligned